sdk/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs/src/Listeners/EventHubsMetricsProvider.cs

```diff
@@ -0,0 +1,194 @@
+        private readonly BlobCheckpointStoreInternal _checkpointStore;
+
+        private  DateTime _nextPartitionLogTime;
+        private  DateTime _nextPartitionWarningTime;
+
+        public EventHubsMetricsProvider(
+            string functionId,
+            IEventHubConsumerClient client,
+            BlobCheckpointStoreInternal checkpointStore,
+            ILoggerFactory loggerFactory)
+        {
+            _functionId = functionId;
+            _client = client;
+            _logger = loggerFactory.CreateLogger<EventHubsMetricsProvider>();
+            _checkpointStore = checkpointStore;
+
+            _nextPartitionLogTime = DateTime.UtcNow;
+            _nextPartitionWarningTime = DateTime.UtcNow;
+        }
+
+        public async Task<EventHubsTriggerMetrics> GetMetricsAsync()
+        {
+            EventHubsTriggerMetrics metrics = new EventHubsTriggerMetrics();
+            string[] partitions = null;
+
+            try
+            {
+                partitions = await _client.GetPartitionsAsync().ConfigureAwait(false);
+                metrics.PartitionCount = partitions.Length;
+            }
+            catch (Exception e)
+            {
+                _logger.LogWarning($"Encountered an exception while checking EventHub '{_client.EventHubName}'. Error: {e.Message}");
+                return metrics;
+            }
+
+            // Get the PartitionRuntimeInformation for all partitions
+            _logger.LogInformation($"Querying partition information for {partitions.Length} partitions.");
+            var partitionPropertiesTasks = new Task<PartitionProperties>[partitions.Length];
+            var checkpointTasks = new Task<EventProcessorCheckpoint>[partitionPropertiesTasks.Length];
+
+            for (int i = 0; i < partitions.Length; i++)
+            {
+                partitionPropertiesTasks[i] = _client.GetPartitionPropertiesAsync(partitions[i]);
+
+                checkpointTasks[i] = _checkpointStore.GetCheckpointAsync(
+                        _client.FullyQualifiedNamespace,
+                        _client.EventHubName,
+                        _client.ConsumerGroup,
+                        partitions[i],
+                        CancellationToken.None);
+            }
+
+            await Task.WhenAll(partitionPropertiesTasks).ConfigureAwait(false);
+            EventProcessorCheckpoint[] checkpoints;
+
+            try
+            {
+                checkpoints = await Task.WhenAll(checkpointTasks).ConfigureAwait(false);
+            }
+            catch
+            {
+                // GetCheckpointsAsync would log
+                return metrics;
+            }
+
+            return CreateTriggerMetrics(partitionPropertiesTasks.Select(t => t.Result).ToList(), checkpoints);
+        }
+        internal EventHubsTriggerMetrics CreateTriggerMetrics(List<PartitionProperties> partitionRuntimeInfo, EventProcessorCheckpoint[] checkpoints, bool alwaysLog = false)
```

Comment:
Please add a new attribute in the EventHubTriggerMetricsClass that tracks the eventCount per partition -- see code pointer for Scale Controller [here](https://msazure.visualstudio.com/One/_git/AAPT-Antares-ScaleController/pullrequest/4947777?path=/src/ScaleController/Microsoft.Azure.AppService.ScaleController.Extension/ActivationTasks/EventHubActivationTask.cs&_a=files)

Reply:
@chiangvincent ok i will change that